lib/utils/display.js

```diff
@@ -355,12 +361,29 @@ class Display {
         // The convention when calling input.read is to pass in a single fn that returns
         // the promise to await. resolve and reject are provided by proc-log
         const [res, rej, p] = args
-        return input.start(() => p()
-          .then(res)
-          .catch(rej)
-          // Any call to procLog.input.read will render a prompt to the user, so we always
-          // add a single newline of output to stdout to move the cursor to the next line
-          .finally(() => output.standard('')))
+
+        // Silent inputs like password prompts require special handling
+        // to preserve output when users hit enter (see input.KEYS.end).
+        this.#silentPrompt = meta[META]?.silentPrompt || false
```

Comment:
This makes me nervous.  The display class is global, and state should be for the entire npm process, not for a single event.  It will likely mean we need to pass this along as we are calling the input events.


It seems here we have a more subtle issue in that `proc-log` itself isn't aware of `META` in the input start/end lifecycle.    It's an oversight and one we likely need to fix there instead.  Then our input handler in the display layer can rely on having the `META` passed into it as originally requested.

TLDR: the real problem here is that we need to ensure that `META` is maintained and passed along through the entire input lifecycle so that `input.KEYS.end` has access to it as originally requested.